packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemOption/ItemOption.js

```diff
@@ -0,0 +1,119 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+
+import { Search } from '@carbon/react';
+
+import { Checkmark } from '@carbon/react/icons';
+
+import PropTypes from 'prop-types';
+import { ConditionBuilderContext } from '../../ConditionBuilderContext/ConditionBuilderProvider';
+import {
+  blockClass,
+  translateWithId,
+} from '../../ConditionBuilderContext/DataConfigs';
+
+export const ItemOption = ({ conditionState = {}, config = {}, onChange }) => {
+  const { popOverSearchThreshold } = useContext(ConditionBuilderContext);
+  const contentRef = useRef();
+
+  const allOptions = config.options;
+  const [filteredItems, setFilteredItems] = useState(config.options);
+  const selection = conditionState.value;
+  useEffect(() => {
+    //this will focus the first input field in the popover
+
+    if (contentRef.current) {
+      const firstFocusableElement =
+        contentRef.current.querySelector('input, button,li');
+
+      if (firstFocusableElement) {
+        firstFocusableElement.focus();
+      }
+    }
+  }, [allOptions]);
+
+  const onClickHandler = (evt, option) => {
+    onChange(option.id, evt);
+  };
+
+  const onSearchChangeHandler = (evt) => {
```

Comment:
operations like filtering rarely need to be stored in state as it can be derived directly from props. this handler should be establishing the value for `Search` and then you can establish filtering in the render. something like

```
const [value, setValue] = useState('');
const { options } = config;
...
const onSearchChangeHandler = (evt) => {
  const { value } = evt.target;
  setValue(value);
};

const filteredOptions = options.filter((option) =>
  option.label.toLowerCase().includes(value.toLowerCase())
);
...
<Search value={value} ... />
```

Reply:
Hey,
i am little confused with your suggestion.
 If we keep introduce `value` as state variable instead of `filteredOptions` as we have now, we may again need to listen to `value` changes (useEffect) to update the filteredOptions.
Now we can manage this with single state `filteredOptions` and reduce a useeffect as well. 
Can you help me I am missing any thoughts?

Reply:
when the state of `value` changes it causes the component to re-render, which then would cause the `filteredOptions` to re-evaluate with the new `value` value. we should be keeping the state of the search value to ensure that it's a controlled component.

Reply:
let me know if you have any follow up questions about this 👍

Reply:
hey , you are right. Made the changes accordingly